atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleDdlTable.java

```diff
@@ -152,14 +152,14 @@ public void drop() {
             // If table does not exist, do nothing
         }
 
-        dropTableMetadataAndClearTableSizeCache();
+        clearTableSizeCacheAndDropTableMetadata();
     }
 
-    private void dropTableMetadataAndClearTableSizeCache() {
+    private void clearTableSizeCacheAndDropTableMetadata() {
+        TableSizeCache.clearCacheForTable(tableRef);
```

Comment:
What are the ramifications of a erroneous cache hit here?  Switching this would result in a danger of an out of date cache if some managed to get a hit in before deleting the metadata table from the backing store.  We may want to consider locking the cache for this time.